renderer/components/ConfigBox.tsx

```diff
@@ -1,77 +1,98 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {
-  Modal,
-  Button,
-  FormGroup,
-  Form,
-  FormControl,
-  ControlLabel,
-} from 'react-bootstrap';
+import { Modal, Button, FormGroup, Form, FormControl, ControlLabel } from 'react-bootstrap';
 import { remote, ipcRenderer } from 'electron';
 import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
 import _ from 'lodash';
 import { getValidationState, logging } from '../utils/utils';
 import { updateFieldControl } from '../actions/FieldActions';
 import { ipChange } from '../actions/InfoActions';
 
-
 const storage = remote.require('electron-json-storage');
 
-class ConfigBoxComponent extends React.Component {
-  constructor(props) {
+interface Config {
+  stationNumber: number;
+  bridgeAddress: string;
+}
+
+interface StateProps {
+  stationNumber: number;
+  ipAddress: string;
+}
+
+interface DispatchProps {
+  onIPChange: (ipAddress: string) => void;
+  onFCUpdate: (config: Config) => void;
+}
+
+interface OwnProps {
+  fcAddress: string;
+  shouldShow: boolean;
+  hide: () => void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+interface State {
+  ipAddress: string;
+  fcAddress: string;
+  stationNumber: number;
+  originalIPAddress: string;
+  originalStationNumber: number;
+  originalFCAddress: string;
+}
+
+class ConfigBoxComponent extends React.Component<Props, State> {
+  constructor(props: Props) {
     super(props);
+
     this.state = {
       ipAddress: this.props.ipAddress,
       fcAddress: this.props.fcAddress,
       stationNumber: this.props.stationNumber,
-      originalIP: this.props.ipAddress,
-      originalSN: this.props.stationNumber,
-      originalFC: this.props.fcAddress,
+      originalIPAddress: this.props.ipAddress,
+      originalStationNumber: this.props.stationNumber,
+      originalFCAddress: this.props.fcAddress,
     };
-    this.saveChanges = this.saveChanges.bind(this);
-    this.disableUploadUpdate = this.disableUploadUpdate.bind(this);
-    this.handleIpChange = this.handleIpChange.bind(this);
-    this.handleFcChange = this.handleFcChange.bind(this);
-    this.handleStationChange = this.handleStationChange.bind(this);
-    this.handleClose = this.handleClose.bind(this);
```

Comment:
@lancejackson @HaileyJang Note the deletion of these `this.*.bind(this)`, we can avoid this by converting these functions to arrow functions which I've done below.